package-drone/src/main/java/de/dentrassi/pm/jenkins/UploaderV3.java

```diff
@@ -148,61 +142,52 @@ public boolean complete ()
                     switch ( response.getStatusLine ().getStatusCode () )
                     {
                         case 200:
-                            processUploadResult ( makeString ( resEntity ) );
-                            return true;
+                            return processUploadResult ( makeString ( resEntity ) );
                         case 404:
-                            this.listener.error ( "Failed to find upload endpoint V3. This could mean that you configured a wrong server URL or that the server does not support the Upload V3. You will need a version 0.14+ of Eclipse Package Drone. It could also mean that you did use wrong credentials." );
-                            return false;
+                            throw new IOException ( "Failed to find upload endpoint V3. This could mean that you configured a wrong server URL or that the server does not support the Upload V3. You will need a version 0.14+ of Eclipse Package Drone. It could also mean that you did use wrong credentials." );
                         default:
-                            if ( !handleError ( response ) )
+                            String errorMessage = "Failed to upload: " + response.getStatusLine ();
+                            String httpResponseErrorMessage = getErrorMessage ( response );
+                            if ( httpResponseErrorMessage != null )
                             {
-                                addErrorMessage ( "Failed to upload: " + response.getStatusLine () );
+                                errorMessage += " " + httpResponseErrorMessage;
```

Comment:
add a new line to keep the previous behaviour `+= "\n" + httpResponseErrorMessage`